include/pmacc/Environment.hpp

```diff
@@ -520,6 +524,11 @@ namespace detail
                 CUDA_CHECK(rc); /*error message*/
             }
         }
+        if(!deviceSelectionSuccessfull)
+        {
+            std::cerr << "Failed to select one of the " << deviceNumber << " devices." << std::endl;
```

Comment:
Why `deviceNumber` here? This variable looks to be the index of a device, not the number of devices (which is `num_gpus`).